src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -0,0 +1,218 @@
+import {useRecoilState} from 'recoil';
+
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import profileAtom from '../../../atom/profileAtom';
+import useChatUtilsHook from '../chat/useChatUtilsHook';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import useUserAuthHook from '../auth/useUserAuthHook';
+import TokenStorage, {ITokenEnum} from '../../../utils/storage/custom/tokenStorage';
+import {getAnonymousUserId, getUserId} from '../../../utils/users';
+
+const usePushNotificationHook = () => {
+  const isIos = Platform.OS === 'ios';
+
+  const navigation = useNavigation();
+  const {signedProfileId, anonProfileId} = useUserAuthHook();
+  const {localDb} = useLocalDatabaseHook();
+  const {fetchChannelDetail, setSelectedChannel} = useChatUtilsHook();
+  const [, setProfileAtom] = useRecoilState(profileAtom);
+
+  const [isLoadingFetchingChannelDetail, setIsLoadingFetchingChannelDetail] = useState(false);
+
+  const __createChannel = () => {
+    PushNotification.createChannel(
+      {
+        channelId: 'bettersosialid', // (required)
+        channelName: 'New Messages & Comments', // (required)
+        playSound: true, // (optional) default: true
+        soundName: 'default', // (optional) See `soundName` parameter of `localNotification` function
+        importance: 4, // (optional) default: 4. Int value of the Android notification importance
+        vibrate: true // (optional) default: true. Creates the default vibration patten if true.
+      },
+      (created) => {
+        if (__DEV__) {
+          console.log(`createChannel returned '${created}'`);
+        }
+      } // (optional) callback returns whether the channel was created, false means it already existed.
+    );
+  };
+
+  const __pushNotifIos = (message) => {
+    const {title, body} = message.notification;
+    PushNotificationIOS.addNotificationRequest({
+      title,
+      body,
+      id: message.messageId,
+      userInfo: message.data
+    });
+  };
+
+  const __pushNotifAndroid = (remoteMessage) => {
+    const {title, body} = remoteMessage.notification;
+    PushNotification.localNotification({
+      id: '123',
+      title,
+      channelId: 'bettersosialid',
+      message: body,
+      data: remoteMessage.data
+    });
+  };
+
+  const __handlePushNotif = (remoteMessage) => {
+    const {data} = remoteMessage;
+    if (data.channel_type !== 3) {
+      if (isIos) {
+        __pushNotifIos(remoteMessage);
+      } else {
+        __pushNotifAndroid(remoteMessage);
+      }
+    }
+  };
+
+  const __updateProfileAtomId = async () => {
+    const signedUserId = await getUserId();
+    const anonUserId = await getAnonymousUserId();
+    const token = TokenStorage.get(ITokenEnum.token);
+    const anonymousToken = TokenStorage.get(ITokenEnum.anonymousToken);
+
+    setProfileAtom({
+      anonProfileId: anonUserId,
+      signedProfileId: signedUserId,
+      token,
+      anonymousToken
+    });
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `__updateProfileAtomId` function, you are using async/await but not handling potential errors with a try/catch block. This could lead to unhandled promise rejections if an error occurs. 

```diff
+   const __updateProfileAtomId = async () => {
+     try {
+       const signedUserId = await getUserId();
+       const anonUserId = await getAnonymousUserId();
+       const token = TokenStorage.get(ITokenEnum.token);
+       const anonymousToken = TokenStorage.get(ITokenEnum.anonymousToken);
+
+       setProfileAtom({
+         anonProfileId: anonUserId,
+         signedProfileId: signedUserId,
+         token,
+         anonymousToken
+       });
+     } catch (error) {
+       console.error('Error updating profile atom id:', error);
+     }
+   };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->